gtsam/inference/doc/BayesTree.ipynb

```diff
@@ -0,0 +1,427 @@
+        },
+        "id": "bayestree_eliminate_code",
+        "outputId": "456789ab-cdef-0123-4567-89abcdef0123"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "Original Factor Graph:\n",
+            "\n",
+            "size: 7\n",
+            "factor 0: \n",
+            "  A[x0] = [\n",
+            "\t-1\n",
+            "]\n",
+            "  b = [ 0 ]\n",
+            "  Noise model: unit (1) \n",
+            "factor 1: \n",
+            "  A[x0] = [\n",
+            "\t-1\n",
+            "]\n",
+            "  A[x1] = [\n",
+            "\t1\n",
+            "]\n",
+            "  b = [ 0 ]\n",
+            "  Noise model: unit (1) \n",
+            "factor 2: \n",
+            "  A[x1] = [\n",
+            "\t-1\n",
+            "]\n",
+            "  A[x2] = [\n",
+            "\t1\n",
+            "]\n",
+            "  b = [ 0 ]\n",
+            "  Noise model: unit (1) \n",
+            "factor 3: \n",
+            "  A[l1] = [\n",
+            "\t-1\n",
+            "]\n",
+            "  A[x0] = [\n",
+            "\t1\n",
+            "]\n",
+            "  b = [ 0 ]\n",
+            "  Noise model: unit (1) \n",
+            "factor 4: \n",
+            "  A[l1] = [\n",
+            "\t-1\n",
+            "]\n",
+            "  A[x1] = [\n",
+            "\t1\n",
+            "]\n",
+            "  b = [ 0 ]\n",
+            "  Noise model: unit (1) \n",
+            "factor 5: \n",
+            "  A[l2] = [\n",
+            "\t-1\n",
+            "]\n",
+            "  A[x1] = [\n",
+            "\t1\n",
+            "]\n",
+            "  b = [ 0 ]\n",
+            "  Noise model: unit (1) \n",
+            "factor 6: \n",
+            "  A[l2] = [\n",
+            "\t-1\n",
+            "]\n",
+            "  A[x2] = [\n",
+            "\t1\n",
+            "]\n",
+            "  b = [ 0 ]\n",
+            "  Noise model: unit (1) \n"
+          ]
+        }
+      ],
+      "source": [
+        "# Create a simple Gaussian Factor Graph (more complex this time)\n",
+        "graph = GaussianFactorGraph()\n",
+        "model = gtsam.noiseModel.Isotropic.Sigma(1, 1.0)\n",
+        "graph.add(X(0), -np.eye(1), np.zeros(1), model)           # Prior on x0\n",
+        "graph.add(X(0), -np.eye(1), X(1), np.eye(1), np.zeros(1), model) # x0 -> x1\n",
+        "graph.add(X(1), -np.eye(1), X(2), np.eye(1), np.zeros(1), model) # x1 -> x2\n",
+        "graph.add(L(1), -np.eye(1), X(0), np.eye(1), np.zeros(1), model) # l1 -> x0 (measurement)\n",
+        "graph.add(L(1), -np.eye(1), X(1), np.eye(1), np.zeros(1), model) # l1 -> x1 (measurement)\n",
+        "graph.add(L(2), -np.eye(1), X(1), np.eye(1), np.zeros(1), model) # l2 -> x1 (measurement)\n",
+        "graph.add(L(2), -np.eye(1), X(2), np.eye(1), np.zeros(1), model) # l2 -> x2 (measurement)\n",
+        "\n",
+        "print(\"Original Factor Graph:\")\n",
```

Comment:
Rather than print use graphviz to show. And immediately show bayes tree after elimination rather than at end?